src/polyhedron.jl

```diff
@@ -0,0 +1,189 @@
+export ConvexHullLib
+
+type ConvexHullLib <: PolyhedraLibrary
+    precision::Symbol
+
+    function ConvexHullLib(precision::Symbol=:float)
+        if !(precision in [:float, :exact])
+            error("Invalid precision, it should be :float or :exact")
+        end
+        new(precision)
+    end
+end
+
+type ConvexHullPolyhedron{N, T} <: Polyhedron{N, T}
+    ine::Nullable{HRepresentation{N, T}}
+    inel::Nullable{LiftedHRepresentation{N, T}}
+    ext::Nullable{VRepresentation{N, T}}
+    extl::Nullable{LiftedVRepresentation{N, T}}
+    hlinearitydetected::Bool
+    vlinearitydetected::Bool
+    noredundantinequality::Bool
+    noredundantgenerator::Bool
+
+    function ConvexHullPolyhedron(ine::HRepresentation{N, T}, ext::VRepresentation{N, T}, hld::Bool, vld::Bool, nri::Bool, nrg::Bool)
+        new(ine, nothing, ext, nothing, hld, vld, nri, nrg)
+    end
+    function ConvexHullPolyhedron(ine::HRepresentation{N, T})
+        new(ine, nothing, nothing, nothing, false, false, false, false)
+    end
+    function ConvexHullPolyhedron(ext::VRepresentation{N, T})
+        new(nothing, nothing, ext, nothing, false, false, false, false)
+    end
+end
+
+# ine may decompose fast but if ine is nothing I do not want to ask to compute it to see the type it is
+# saying false normally do not give troubles
+decomposedhfast{N, T}(::Type{ConvexHullPolyhedron{N, T}}) = false
+decomposedvfast{N, T}(::Type{ConvexHullPolyhedron{N, T}}) = false
+decomposedhfast{N, T}(p::ConvexHullPolyhedron{N, T}) = decomposedhfast(ConvexHullPolyhedron{N, T})
+decomposedvfast{N, T}(p::ConvexHullPolyhedron{N, T}) = decomposedvfast(ConvexHullPolyhedron{N, T})
+
+# Helpers
+function getine(p::ConvexHullPolyhedron)
+    if isnull(p.ine)
+        if !isnull(p.inel)
+            p.ine = p.inel
```

Comment:
Cool, thanks for the explanation. I was confused by the naming, but the abstraction does make sense.